src/proofnetwork/baserestapi.cpp

```diff
@@ -282,6 +287,58 @@ bool BaseRestApiPrivate::replyShouldBeHandledByError(QNetworkReply *reply) const
     return (errorHundreds != 2 && errorHundreds != 4);
 }
 
+std::tuple<QString, NetworkErrorCode::Code> BaseRestApiPrivate::errorByCheckConnection(QNetworkReply *reply)
```

Comment:
Why don't return a filled Failure here (with network module code and user friendly hint since it looks like we always set them for those errors)? Anyway it should be something like buildReplyFailure or calcReplyFailure (to indicate that it also does some extra stuff and can be blocking)